### DISCUSSION TITLE
problem in getOrCreateAnvilEthConfig() function

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#676] [General]
i am not getting the logic behind these line
```
 if (activeNetworkConfig.wethUsdPriceFeed != address(0)) {
            return activeNetworkConfig;
        }
```

**Comment by Tanmay53** [comment]
Here, the code checks if the price feed is already filled or not.

Since, Solidity by default assigns an empty address to an address type variable (0x00....000).

Therefore, the condition checks if `wethUsdPriceFeed` is not equal to address(0) i.e. 0x000....000 .

**Reply by khushbusandhu** [reply]
Why to check that since we are a new creating a function for anvil config then how can pricefeed be also filled

**Reply by 0xusmanf** [reply]
Can you please share the full file, it is hard to understand or explain the context by just looking at the given code.

**Reply by Tanmay53** [reply]
This functionality will be helpful when we use the HelperConfig multiple time in script, then it will help in not deploying the pricefeed Mock again.

**Comment by theirrationalone** [comment]
> Why to check that since we are a new creating a function for anvil config then how can pricefeed be also filled


Yeah, of course, Pricefeed will get filled only when we create anvil config by calling `createOrGetAnvilNetworkConfig` but What for next???
I mean what if we again make a call to `createOrGetAnvilNetworkConfig` function then in that case we don't need to create another new configuration for anvil we can use previously created configuration. That's Why that if check is there and guarding if we have pricefeed address already then we don't need to create another new configuration. Therefore, We can use our previously created one.

Hope you get it.

**Reply by khushbusandhu** [reply]
yes now I get it 
sorry for the late response actually on Monday and Tuesday I have classes till 6 pm that's why couldn't able to check it
